File: SpringBootMultipleMavenModules/domain/src/main/java/rc/domain/Hotel.java
package rc.domain;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
public class Hotel {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private long id;
    private String name;
    private int classification;
    private boolean isOpen;

    protected Hotel(){}

    public Hotel(String name, int classification, boolean isOpen){
        this.name = name;
        this.classification = classification;
        this.isOpen = isOpen;
    }

    public String getName() {
        return name;
    }

    public int getClassification() {
        return classification;
    }

    public long getId() {
        return id;
    }

    public boolean isOpen() {
        return isOpen;
    }
}

File: SpringBootKotlinDemo/src/main/kotlin/rc/demo/Hotel.kt
package rc.demo

import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id

@Entity
class Hotel(val name: String, val classification: Int, val nbRooms: Int) {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long = 0
    var nbFreeRooms: Int = this.nbRooms

    /**
     * We need a default constructor for this class because it is serialized/deserialzied
     * by the REST controller
     */
    constructor(): this("", 0, 0)

    fun checkIn(nbGuests: Int) {
        if (this.nbFreeRooms >= nbGuests) {
            this.nbFreeRooms -= nbGuests
        }
    }

    fun checkOut(nbGuests: Int) {
        this.nbFreeRooms += nbGuests
    }
}